scenarios/sns_secrets/terraform/api_gateway.tf

```diff
@@ -0,0 +1,92 @@
+resource "aws_api_gateway_rest_api" "cg_api" {
+  name        = "cg-api-${var.cgid}"
+  description = "API for demonstrating leaked API key scenario"
+  tags = {
+    Name     = "cg-api-${var.cgid}"
+    Stack    = var.stack-name
+    Scenario = var.scenario-name
+  }
+}
+
+resource "aws_api_gateway_resource" "cg_resource" {
+  rest_api_id = aws_api_gateway_rest_api.cg_api.id
+  parent_id   = aws_api_gateway_rest_api.cg_api.root_resource_id
+  path_part   = "resource-${var.cgid}"
+}
+
+resource "aws_api_gateway_method" "cg_method" {
+  rest_api_id   = aws_api_gateway_rest_api.cg_api.id
+  resource_id   = aws_api_gateway_resource.cg_resource.id
+  http_method   = "GET"
+  authorization = "NONE"
+  api_key_required = true
+}
+
+resource "aws_api_gateway_integration" "cg_integration" {
+  rest_api_id = aws_api_gateway_rest_api.cg_api.id
+  resource_id = aws_api_gateway_resource.cg_resource.id
+  http_method = aws_api_gateway_method.cg_method.http_method
+  type        = "MOCK"
+  request_templates = {
+    "application/json" = "{\"statusCode\": 200}"
+  }
+}
+
+resource "aws_api_gateway_method_response" "cg_method_response" {
+  rest_api_id = aws_api_gateway_rest_api.cg_api.id
+  resource_id = aws_api_gateway_resource.cg_resource.id
+  http_method = aws_api_gateway_method.cg_method.http_method
+  status_code = "200"
+}
+
+resource "aws_api_gateway_integration_response" "cg_integration_response" {
+  rest_api_id = aws_api_gateway_rest_api.cg_api.id
+  resource_id = aws_api_gateway_resource.cg_resource.id
+  http_method = aws_api_gateway_method.cg_method.http_method
+  status_code = aws_api_gateway_method_response.cg_method_response.status_code
+  response_templates = {
+    "application/json" = <<EOF
+    {
+      "message": "Access granted.",
+      "user_data": {
+        "user_id": "1337",
+        "username": "SuperAdmin",
+        "email": "SuperAdmin@notarealemail.com",
+        "password": "p@ssw0rd123"
+      },
+      "final_flag": "FLAG{SNS_S3cr3t$_ar3_FUN}"
```

Comment:
Weird output with the flag being returned as `FLAG{SNS_S3cr3t}`

Other than that works as expected!

Reply:
```
curl -X GET -H "x-api-key: 45a3da610dc64703b10e273a4db135bf" https://y9lotdavd4.execute-api.us-east-1.amazonaws.com/prod-sns_secrets_cgidi6dsa21cut/resource-sns_secrets_cgidi6dsa21cut 
    {
      "message": "Access granted.",
      "user_data": {
        "user_id": "1337",
        "username": "SuperAdmin",
        "email": "SuperAdmin@notarealemail.com",
        "password": "p@ssw0rd123"
      },
      "final_flag": "FLAG{SNS_S3cr3t}"
    }
```

Reply:
Oh strange! Oh well, it still gives a flag :D -- Thanks for all the feedback, comments, and suggestions on this!